skglm/solvers/multitask_bcd_solver.py

```diff
@@ -254,32 +253,19 @@ def bcd_solver(
                     ws)
             else:
                 _bcd_epoch(X, Y, W, XW, datafit, penalty, ws)
+
             if use_acc:
-                last_K_w[epoch % (K + 1)] = W[ws, :].ravel()
-
-                # 3) do Anderson acceleration on smaller problem
-                if epoch % (K + 1) == K:
-                    for k in range(K):
-                        U[k] = last_K_w[k + 1] - last_K_w[k]
-                    C = np.dot(U, U.T)
-
-                    try:
-                        z = np.linalg.solve(C, np.ones(K))
-                        c = z / z.sum()
-                        W_acc = np.zeros((n_features, n_tasks))
-                        W_acc[ws, :] = np.sum(
-                            last_K_w[:-1] * c[:, None], axis=0).reshape(
-                                (ws_size, n_tasks))
-                        p_obj = datafit.value(Y, W, XW) + penalty.value(W)
-                        Xw_acc = X[:, ws] @ W_acc[ws]
-                        p_obj_acc = datafit.value(
-                            Y, W_acc, Xw_acc) + penalty.value(W_acc)
-                        if p_obj_acc < p_obj:
-                            W[:] = W_acc
-                            XW[:] = Xw_acc
-                    except np.linalg.LinAlgError:
-                        if max(verbose - 1, 0):
-                            print("----------Linalg error")
+                W_acc, XW_acc = accelerator.extrapolate(W.ravel(), XW.ravel())
+
+                p_obj = datafit.value(Y, W, XW) + penalty.value(W)
+                p_obj_acc = (datafit.value(Y, W_acc.reshape(-1, n_tasks),
+                                           XW_acc.reshape(-1, n_tasks))
```

Comment:
@mathurinm, I m not sure whether I pass in a view or a copy.

Link to [reshape Numpy doc ](https://numpy.org/doc/stable/reference/generated/numpy.reshape.html#:~:text=Returns,the%20returned%20array.)